### PR TITLE
Replace lib_phpQuery by PhpGt/CssXPath

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/PhpGt/CssXPath
Much lighter and cleaner

### CODE REVIEW

File: app/Models/Entry.php [hunk 1]
```diff
@@ -480,40 +480,37 @@ public function isDay(int $day, int $today): bool {
...
+			$doc->loadHTML($html, LIBXML_NONET | LIBXML_NOERROR | LIBXML_NOWARNING);
+			$xpath = new DOMXPath($doc);
 
 			if ($maxRedirs > 0) {
 				//Follow any HTML redirection
-				/**
-				 * @var phpQueryObject @metas
-				 */
-				$metas = $doc->find('meta[http-equiv][content]');
+				$metas = $xpath->query('//meta[@content]');
```
Comment (Frenzie) [inline app/Models/Entry.php:489]: Does this deal with random whitespace that doesn't affect HTML?

Comment (Alkarex) [reply app/Models/Entry.php:489]: Yes

**Comment by Alkarex** [comment]
Ping @marienfressinaud and @aledeg : if you use full content retrieval for some of your feeds, it would be nice if you could give this a try

**Comment by Alkarex** [comment]
@javerous You did a a great work regarding full content retrieval https://github.com/FreshRSS/FreshRSS/pull/2778 ; so if you are around, it would be nice if you could give this PR a try, as there are risks of regressions :-)
And if you have not seen it, you might be interested in this slightly related feature, which freshly landed in the edge branch: https://github.com/FreshRSS/FreshRSS/pull/4220

**Comment by math-GH** [comment]
First glance it looks fine. Did not checked the code.

File: phpstan.neon [hunk 2]
```diff
@@ -7,7 +7,6 @@ parameters:
 	paths:
 		- .
 	excludePaths:
-		- lib/lib_phpQuery.php
```
Comment (math-GH) [inline phpstan.neon:10]: should CssXPath be excluded from phpstan scan?

Comment (Frenzie) [reply phpstan.neon:10]: Exclusion doesn't seem necessary if it's not causing issues, but potentially fair point from a handful of wasted cpu cycles.

Comment (Alkarex) [reply phpstan.neon:10]: For now, I would rather keep it (phpstan uses some caching) to be informed in case of regression

**Comment by marienfressinaud** [comment]
> Ping @marienfressinaud and @aledeg : if you use full content retrieval for some of your feeds, it would be nice if you could give this a try

For the few I had, it seems to work well :+1: (I prefer your approach rather than previous lib_phpQuery btw :))

**Comment by Alkarex** [comment]
> For the few I had, it seems to work well 👍 (I prefer your approach rather than previous lib_phpQuery btw :))

Cool :-) Let's merge then

**Comment by javerous** [comment]
> @javerous You did a a great work regarding full content retrieval #2778 ; so if you are around, it would be nice if you could give this PR a try, as there are risks of regressions :-)
> And if you have not seen it, you might be interested in this slightly related feature, which freshly landed in the edge branch: #4220

Hi @Alkarex - sorry for the late reply.

Thank you for your comment. I tested it a bit (even if the feature has been merged since a long time now…), I didn't notice any issues with the selectors I'm using !

The "HTML+XPath (Web scraping)" feature seems really cool ! I tested it a bit, it's very nice 👏 (and well, it may also deserve some way to validate / preview what was entered, in the future - it took me some time to have what I wanted 😅)

**Comment by Alkarex** [comment]
Thanks for the feedback @javerous :-)

**Pull request merged**